### PR TITLE
Add a "/vtm [format]" command for standalone team validation

### PR DESCRIPTION
**Pull request by sobolews**
The /validate command allows users to validate their team for its format as part
of the teambuilding experience. Currently, the only way to validate a team is to
actually try to battle with it, then go back and forth from the teambuilder
until the team is not rejected. User#validateTeam will be made accessible via a
new button in the teambuilder.

Screenshots: http://imgur.com/a/dVj3F

### CODE REVIEW

**Comment by Slayer95** [comment]
We probably don't need the ``user.validateTeam`` boilerplate

**Comment by Zarel** [comment]
Yeah, the `TeamBuilder.validateTeam` call can be directly in commands.

We should probably increment the ResourceMonitor battle counter for this command, though – team validation is pretty server-intensive.

We should also probably rename this command to something less obvious. e.g. `/vtm`. The command can only be used after a `/utm` call so it shouldn't be easy for a user to call if they don't know what it's doing.

**Comment by Slayer95** [comment]
Also make sure that the format exists.

**Comment by Rhythms88** [comment]
If the format doesn't exist shouldn't it just default to none?

**Comment by Slayer95** [comment]
What does it mean to validate a team in no format, @Rhythms88 ?

**Comment by Zarel** [comment]
We could default to AG. In other words, validate for legality but not for a specific format's rules.

**Comment by Slayer95** [comment]
Oh, right. AG is now truly AG after Endless Battle Clause rework.

**Comment by sobolews** [comment]
If the format is set to "(none)" for the team, it currently just tells the user to pick a format. Would you prefer it to validate for AG instead?

**Comment by sobolews** [comment]
Changed /validate to /vtm, and added AG as the default format.

**Comment by Zarel** [comment]
Can you add the ResourceMonitor count?

**Comment by sobolews** [comment]
Sure, should a validate call count as a battle, prepBattle, or should it have a separate bucket? Also, what does the ResourceMonitor.battles counter do exactly? Is it consumed by something offline? I can't see anything that consumes that attribute.

File: commands.js [hunk 1]
```diff
@@ -2183,6 +2183,18 @@ var commands = exports.commands = {
 		user.team = target;
 	},
 
+	vtm: function (target, room, user, connection) {
+		var format = Tools.getFormat(target);
+		if (!format.name) format = Tools.getFormat('Anything Goes');
```
Comment (Slayer95) [inline commands.js]: ``if (format.effectType !== 'Format')``

Comment (Slayer95) [inline commands.js]: Also another existence check should be done afterwards, for servers could remove the Anything Goes format.

**Comment by Zarel** [comment]
Team validation is what prepBattle counts.

It's possible the counter isn't used for anything.

**Comment by sobolews** [comment]
Addressed @Slayer95's comments and added a countPrepBattle call.

File: commands.js [hunk 2]
```diff
@@ -2183,6 +2183,26 @@ var commands = exports.commands = {
 		user.team = target;
 	},
 
+	vtm: function (target, room, user, connection) {
+		if (ResourceMonitor.countPrepBattle(connection.ip || connection.latestIp, user.name)) {
```
Comment (Zarel) [inline commands.js]: Remove ` || connection.latestIp`, it's meaningless.

**Comment by sobolews** [comment]
Removed ```|| connection.latestIp```.

**Comment by Zarel** [comment]
Looks good.

**Pull request merged**